indexd/index/drivers/alchemy.py

```diff
@@ -37,6 +40,94 @@
 Base = declarative_base()
 
 
+def _get_authorized_resources(requested_method="READ") -> list[str]:
+    """Retrieve the user's authorized resources from Arborist."""
+    # Token is __not__ required for RBAC, but we can use it to get the user's context.
+    # See https://github.com/uc-cdis/indexd/pull/400#discussion_r2243298256
+    # `Arborist allows no token to be sent on purpose, it allows assignment of anonymous access. So we don't want to raise this error here`
+    # if not flask.request.authorization:
+    #     raise AuthError("Authorization header is required for RBAC")
+    # token = flask.request.headers.get("Authorization")
+    # if token is None:
+    #     raise AuthError("Authorization header is required for RBAC")
+    authorized_resources = []
+    for resource, permissions in auth.resources().items():
+        for permission in permissions:
+            method = permission['method']
+            service = permission['service']
+            if requested_method == "READ":
+                if "read-storage" == method or service == "indexd":
+                    authorized_resources.append(resource)
+            else:
+                if method in ["create", "update", "delete"]:
+                    authorized_resources.append(resource)
+    return authorized_resources
+
+
+def _check_user_access(authorized_resources: list[str], authz: list[str]) -> None:
+    """Raise an AuthzError if the user is not authorized for the given resources."""
+    for resource in authz:
+        if resource not in authorized_resources:
+            raise AuthzError(
+                f"User is not authorized for project {resource} {authz} {type(authz)}"
```

Comment:
see other comments, let's put the details here into a debug log and leave the authzerror opaque. `User is not authorized`

Reply:
done